src/globalInfo/accessors/aliasAccessor.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Nullable } from '@salesforce/ts-types';
+import { SfdxError } from '../../sfdxError';
+import { GlobalInfo } from '../globalInfoConfig';
+import { SfAliases, SfOrg, SfInfoKeys, SfToken } from '../types';
+
+export type Aliasable = string | Partial<SfOrg> | Partial<SfToken>;
+
+export class AliasAccessor {
+  public constructor(private globalInfo: GlobalInfo) {}
+
+  /**
+   * Returns all the aliases for all the usernames
+   */
+  public getAll(): SfAliases;
+  /**
+   * Returns all the aliases for a given entity
+   *
+   * @param entity the aliasable entity that you want to get the aliases of
+   */
+  public getAll(entity: Aliasable): string[];
+  public getAll(entity?: Aliasable): string[] | SfAliases {
+    const all = this.globalInfo.get(SfInfoKeys.ALIASES) || {};
```

Comment:
@peternhale No I think `||` is better in this scenario since `??` will only use the right hand-side operand if the left hand-side is `null` or `undefined`. While it's extremely unlikely, if `this.globalInfo.get(SfInfoKeys.ALIASES)` resolved to an empty string `''` and we used `??` then we'd end up with that empty string instead of the empty object - which would have bad downstream effects.